### ISSUE TITLE
Code clean up and more explanations

### ISSUE THREAD

**Issue by JamesCallanan** [JamesCallanan/Grad-CAM#1]
I think the code is not easily accessible to people other than myself.

It took me a while to figure out how I was computing the heatmap loss from the Experiment 1/Heatmap loss jupyter notebook and I wrote it!

I need to add more explanations throughout the jupyter notebook and remove any redundant code.

e.g. Could include some of the following comments in the Heatmap Loss.ipynb notebook ...

Could also record a video showing how things are calculated. Would be handier. Suspect people would be more likely to watch a video explanation of me running through the Jupyter notebook.

Could pursue this further?

'''
For model training with the tuner.search() method, you need to pass the training images and their corresponding labels as parameters.

As opposed to passing in a single class label for each MRI cross section, I pass in a 2D array. This 2D array consists of 

1) A downsized 2D mask showing where the model shouldn't look (i.e. a mask of the region outside the heart).
This is downsized so it has the same dimensionality as the HiResCAM heatmap. i.e. the same dimensionality as the output from the last convolutional layer.

2) A column which contains the class label for each MRI cross section.

Later I define a custom loss function. This loss function consists of two parts, a heatmap loss component and an MSE loss component.

I can retrieve the mask and the class label for each MRI cross section from the previously described labels that I pass to the tuner.search() method. These are needed in order to calculate the heatmap loss and the MSE loss respectively.

Training is only working with a batch size of 1 currently. Perhaps, it would be worth getting it to work with larger batch sizes. Is a batch size of 1 very inefficient (lots of moving of data from CPU RAM to GPU RAM?

'''
